### PR TITLE
[feat][ci] enable semantic pull request title check

### PR DESCRIPTION
**Pull request by tisonkun**
This is blocked by the INFRA ticket: https://issues.apache.org/jira/browse/INFRA-23751.

Also, as pulsar-client-cpp and blogs moved out, I remove the following scopes:

* cpp
* py
* blog

And change "java" to "client" since "client" is now unambiguous.

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:https://github.com/tisonkun/pulsar/pull/6

### CODE REVIEW

**Comment by nodece** [comment]
I suggest you should use the `amannn/action-semantic-pull-request@v4`. Don't depend on outside service.

**Comment by tisonkun** [comment]
@nodece It can work. Let me file an INFRA issue to add the action to the whitelist.

**Comment by tisonkun** [comment]
@nodece the action can have some security issue: https://issues.apache.org/jira/browse/INFRA-21273 Please check it out.

**Comment by tisonkun** [comment]
@nodece It seems `amannn/action-semantic-pull-request@v4` already be able to use. Could you help with labeling `ready-for-test` or approve this PR to proceed CI?

As you can see, this action tests the case: https://github.com/apache/pulsar/actions/runs/3213505677/jobs/5253224953

File: .github/workflows/ci-semantic-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+name: Semantic Pull Request
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - edited
+      - synchronize
```
Comment (tisonkun) [inline .github/workflows/ci-semantic-pull-request.yml:27]:
```suggestion
```

File: .github/workflows/ci-semantic-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+# under the License.
+
+
+name: Semantic Pull Request
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
```
Comment (tisonkun) [inline .github/workflows/ci-semantic-pull-request.yml:25]:
```suggestion
```

File: .github/workflows/ci-semantic-pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        with:
+          # Type abbreviation comments:
+          # feat -> feature
+          types: |
+            feat
+            fix
+            improve
+            refactor
+            revert
+          # Scope abbreviation comments:
```
Comment (nodece) [inline .github/workflows/ci-semantic-pull-request.yml:46]:
```suggestion
          cleanup
          # Scope abbreviation comments:
```

Comment (tisonkun) [reply .github/workflows/ci-semantic-pull-request.yml:47]: Forced pushed to add it. Also I'm trying to sort the candidate :)

Interesting I lost this candidate when copy-paste.

**Comment by tisonkun** [comment]
cc @lhotari

File: .github/workflows/ci-semantic-pull-request.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          # fn      -> Pulasr Functions
+          # io      -> Pulsar Connectors
+          # offload -> tiered storage
+          # sec     -> security
+          # sql     -> Pulsar Trino Plugin
+          # txn     -> transaction
+          # ws      -> websocket
+          scopes: |
+            admin
+            broker
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply .github/workflows/ci-semantic-pull-request.yml:58]: Please read the PR description:

> Also, as pulsar-client-cpp and blogs moved out, I remove the following scopes:

**Comment by tisonkun** [comment]
You can check the latest semantic pull request check at: https://github.com/apache/pulsar/actions/runs/3213531600/jobs/5253269029

... with name "Semantic Pull Request / Check pull rrequest title"

**Review by nodece — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Comment by Anonymitaet** [comment]
> Also, as pulsar-client-cpp and blogs moved out, I remove the following scopes:
> 
> * cpp
> * py
> * blog
> 
> And change "java" to "client" since "client" is now unambiguous.

Thanks! I've updated the [guide](https://docs.google.com/document/d/1d8Pw6ZbWk-_pCKdOmdvx9rnhPiyuxwq60_TrD68d7BA/edit#bookmark=id.y8943h392zno) accordingly.

File: .github/workflows/ci-semantic-pull-request.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+name: Semantic Pull Request
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - edited
+      - synchronize
```
Comment (nicoloboschi) [inline .github/workflows/ci-semantic-pull-request.yml:27]: @tisonkun the github docs says
```
synchronize: Triggered when a pull request's head branch is updated. For example, when the head branch is updated from the base branch, when new commits are pushed to the head branch, or when the base branch is changed.
```
This means that for every push on the contributor branch, on every commit on the master branch this action is triggered again unnecessarly.

As you know we're facing GH runners supply issues so I suggest to remove this trigger type

Comment (tisonkun) [reply .github/workflows/ci-semantic-pull-request.yml:27]: Make sense. I notice some weird check status not report issue previously when I removed it at https://github.com/apache/pulsar/pull/17961#discussion_r990756680.

But given the functionality is ready now we can try to apply the optimization.

Comment (tisonkun) [reply .github/workflows/ci-semantic-pull-request.yml:27]: So here it is: https://github.com/apache/pulsar/pull/17997

**Pull request merged**